### PR TITLE
Migrate Briefs to GOVUK Frontend V3

### PR DESCRIPTION
**Pull request by domoscargin**
https://trello.com/c/VhmOUlKs/211-3-migrate-briefs-frontend-to-govuk-frontend-v3

This PR updates GOVUK Frontend to version 3 and removes legacy dependencies and assets.

### CODE REVIEW

**Comment by lfdebrux** [comment]
@domoscargin I've added a hacky feature flag in alphagov/digitalmarketplace-content-loader#105 to account for the error hrefs, and opened a separate PR against this repo to migrate to LandRegistry/govuk-frontend-jinja (#354)

**Comment by lfdebrux** [comment]
@domoscargin after trying to do so for this app, I don't think we're ready to move to LandRegistry/govuk-frontend-jinja; there are still lots of instances in digitalmarketplace-govuk-frontend where the code evaluates differently between Jinja and Nunjucks.

We should instead publish a new version of alphagov/govuk-frontend-jinja to support the things we need for this app.

**Comment by domoscargin** [comment]
> @domoscargin after trying to do so for this app, I don't think we're ready to move to LandRegistry/govuk-frontend-jinja; there are still lots of instances in digitalmarketplace-govuk-frontend where the code evaluates differently between Jinja and Nunjucks.
> 
> We should instead publish a new version of alphagov/govuk-frontend-jinja to support the things we need for this app

Thanks for looking into it. I'm just wrangling a few more app tests, but I think so far the only change this needs is https://github.com/alphagov/govuk-frontend-jinja/pull/54. On that PR, I accidentally updated setup.py, but didn't release.

I _think_ that's the only issue causing failing tests at least - haven't dug around manually.

**Comment by lfdebrux** [comment]
101 lines added, 684 lines deleted

_chef kiss_ 👨‍🍳

**Comment by lfdebrux** [comment]
### My review checklist

I've:

- [ ] looked at the code
- [ ] tested locally
- [ ] run functional tests locally
- [ ] run visual regression tests locally

File: gulpfile.js [hunk 1]
```diff
@@ -90,7 +88,23 @@ gulp.task('clean:css', function () {
...
 })
 
-gulp.task('clean', gulp.parallel('clean:js', 'clean:css'))
+gulp.task('clean:legacy', function () {
+  return del(
+    [dmToolkitScssRoot, dmToolkitTemplateRoot, govukCopiedScssRoot]
+  ).then(function (paths) {
+    console.log('💥  Deleted the following legacy directories:\n', paths.join('\n'))
+  })
+})
```
Comment (lfdebrux) [inline gulpfile.js:97]: 💥 🙌

File: app/assets/scss/_iframe_preview.scss [hunk 2]
```diff
@@ -43,4 +43,12 @@ $app-mobile-border-bottom-width: 68px;
...
       width: 30%;
   }
+
+  h2 span:nth-child(1) {
+    @include govuk-font($size: 24, $weight: bold);
+  }
+
+  h2 span:nth-child(2) {
+    @include govuk-font($size: 19, $weight: bold);
+  }
```
Comment (lfdebrux) [inline app/assets/scss/_iframe_preview.scss:53]: What do these do?

Comment (domoscargin) [reply app/assets/scss/_iframe_preview.scss:53]: The preview app stat blocks are a bit custom, so weren't inheriting the page font style. These solve that for that iFrame. I can do some more investigating to work out why it's not inheriting, but thought I'd get it looking right initially.

Screenshot 2020-09-09 at 15 25 35

File: package.json [hunk 3]
```diff
@@ -17,8 +17,7 @@
     "gulp-sass": "^4.0.2",
     "gulp-shell": "^0.7.1",
     "gulp-sourcemaps": "^2.6.5",
-    "gulp-uglify": "^3.0.2",
-    "jquery": "^3.5.0"
```
Comment (lfdebrux) [inline package.json:24]: 🎉

**Review by lfdebrux — APPROVED** [review]
This looks good to me, amazing work. Let's merge it!

**Comment by katstevens** [comment]
Brilliant stuff! Will the functional tests need updating?

**Comment by lfdebrux** [comment]
> Brilliant stuff! Will the functional tests need updating?

When I ran them locally they passed without issue, so I don't think so.

**Review by katstevens — APPROVED** [review]
I love the diff numbers on this so much! 🔥

**Pull request merged**